tests/FakeItEasy.Specs/SelfInitializedFakesSpecs.cs

```diff
@@ -91,48 +98,56 @@ public static class SelfInitializedFakesSpecs
         {
-            "establish"
+            "Given a path that does not exist"
+                .x(() => fileRecorderPath = Path.Combine(Path.GetTempPath(), Guid.NewGuid().ToString()));
+
+            "And a real service to wrap while recording"
                 .x(() =>
-                    {
-                        fileRecorderPath = Path.Combine(Path.GetTempPath(), Guid.NewGuid().ToString());
+                {
+                    realServiceWhileRecording = A.Fake<ILibraryService>();
 
-                        realServiceWhileRecording = A.Fake<ILibraryService>();
-                        realServiceDuringPlayback = A.Fake<ILibraryService>();
+                    A.CallTo(() => realServiceWhileRecording.GetCount("8"))
+                        .Returns(0x8);
+                });
 
-                        A.CallTo(() => realServiceWhileRecording.GetCount("8")).Returns(8);
-                    });
+            "And a real service to wrap while playing back"
+                .x(() => realServiceDuringPlayback = A.Fake<ILibraryService>());
 
-            "when self initializing a fake with a file recorder"
+            "When I use a self-initialized fake recording to the path to get the count for book 8"
                 .x(() =>
+                {
+                    try
                     {
-                        try
-                        {
-                            using (var recorder = Recorders.FileRecorder(fileRecorderPath))
-                            {
-                                var fakeService = A.Fake<ILibraryService>(options => options
-                                    .Wrapping(realServiceWhileRecording).RecordedBy(recorder));
-                                countWhileRecording = fakeService.GetCount("8");
-                            }
-
-                            using (var recorder = Recorders.FileRecorder(fileRecorderPath))
-                            {
-                                var playbackFakeService = A.Fake<ILibraryService>(options => options
-                                    .Wrapping(realServiceDuringPlayback).RecordedBy(recorder));
-
-                                countDuringPlayback = playbackFakeService.GetCount("8");
-                            }
-                        }
-                        finally
+                        using (var recorder = Recorders.FileRecorder(fileRecorderPath))
                         {
-                            File.Delete(fileRecorderPath);
+                            var fakeService = A.Fake<ILibraryService>(options => options
+                                .Wrapping(realServiceWhileRecording).RecordedBy(recorder));
+                            countWhileRecording = fakeService.GetCount("8");
                         }
-                    })
+                    }
+                    catch
+                    {
+                        File.Delete(fileRecorderPath);
+                    }
+                })

```

Comment:
Just to make sure I got this right:
- you split this step into 2 different steps
- so you no longer delete the file in a finally, because it still needs to be there for the next step
- it will still be deleted at the end of the scenario because of the `Teardown`
- you delete it in case something throws

Questions:
- Is the `catch` really necessary? We don't expect an exception to be thrown, so if it does, we want the step to fail, and the `catch` will prevent that. Shouldn't it at least rethrow the exception?
- Deleting in the `catch` is redundant, since the `Teardown` will take care of it